src/solvers/base_solver.cpp

```diff
@@ -29,26 +29,25 @@ BaseSolver::BaseSolver(MPI_Comm comm, int n) : BaseSolver(comm)
   m_gf_initialized.assign(n, false);
 }
 
-void BaseSolver::SetEssentialBCs(const std::vector<int> &                 ess_bdr,
+void BaseSolver::SetEssentialBCs(const std::set<int> &                 ess_bdr,
                                  std::shared_ptr<mfem::VectorCoefficient> ess_bdr_vec_coef,
                                  mfem::ParFiniteElementSpace &fes, int component)
 {
   auto bc = std::make_shared<BoundaryCondition>();
 
-  bc->markers.SetSize(ess_bdr.size());
-
-  for (unsigned int i = 0; i < ess_bdr.size(); ++i) {
-    bc->markers[i] = ess_bdr[i];
-
-    if (bc->markers[i] == 1) {
-      for (auto &existing_bc : m_ess_bdr) {
-        if (existing_bc->markers[i] == 1) {
-          mfem::mfem_warning("Multiple definition of essential boundary! Using first definition given.");
-          bc->markers[i] = 0;
-          break;
-        }
+  bc->markers.SetSize(m_state.front()->mesh->bdr_attributes.Max());
+  bc->markers = 0;
+
+  for (int attr : ess_bdr) {
+    MFEM_ASSERT(attr <= bc->markers.Size(), "Attribute specified larger than what is found in the mesh.");
+    bc->markers[attr-1] = 1;
+    for (auto &existing_bc : m_ess_bdr) {
```

Comment:
I'm not sure how big the meshes get, but depending on the number of boundary conditions/attributes, the solver could also store a set of attributes already assigned to a boundary condition and avoid the linear search here.  

It might also worth be comparing `std::set` and `std::unordered_set` for the boundary attribute indications, but this also depends on how large a "real" mesh is.

Reply:
These will typically be extremely small. However, this is really a stop gap measure until we have a real "entity set" type of boundary marker. The problem here is that the internal MFEM mesh format can only have one attribute marker or "color" per boundary face. In complex multiphysics simulations, you can easily have overlapping areas for different boundary conditions. Production codes use face or nodesets to get around this. Development of this capability is logged in this [issue](https://github.com/LLNL/serac/issues/12).